package/skylark_upload_daemon/src/skylark_upload_daemon.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Swift Navigation Inc.
+ * Contact: Mark Fine <mark@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <curl/curl.h>
+#include <stdlib.h>
+#include <unistd.h>
+
+#include <libskylark.h>
+
+#define VERBOSE
+
+//
+// Upload Daemon - connects to Skylark and sends SBP messages.
+//
+
+static size_t upload_callback(void *p, size_t size, size_t n, void *up)
+{
+  int *fd = (int *)up;
+  ssize_t m = read(*fd, p, size * n);
+  if (m < 0) {
+    return CURL_READFUNC_ABORT;
+  }
+  return m;
+}
+
+int main(void)
```

Comment:
TODO: Add comments here.
TODO: Named fifo should come through command line argument
TODO: Signal handling?
TODO: Handle error codes and log messages.
TODO: Implement settings for enable/URL
TODO: Update exit code